bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -296,88 +328,92 @@ public void onValueReceived(@Nullable String variable, @Nullable String value, @
         } else {
             String binaryState = stateMap.get("BinaryState");
             if (binaryState != null) {
-                State state = binaryState.equals("0") ? OnOffType.OFF : OnOffType.ON;
```

Comment:
Still not sure why 3.2 version worked, but something is wrong here:

```
        if ("insight".equals(getThing().getThingTypeUID().getId())) {
            String insightParams = stateMap.get("InsightParams");

            if (insightParams != null) {
                // ...
            }
        } else {
            String binaryState = stateMap.get("BinaryState");
            if (binaryState != null) {
```

My Insight has id "insight", so will get into first block. But when actually receiving **BinaryState**, not **InsightParams**, this will not be processed. Instead existing **InsightParams** is processed, but this isn't updated. So state will be set to ON although it's actually OFF.

Reply:
Hmm, I will check later.....

Reply:
With latest changes 7db23e7bb88f9125f329bf402bbf2895afe6d84a I now get this:

```
2022-01-21 23:30:35.824 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Received pair 'EnergyPerUnitCost':'4|2000|4' (service 'basicevent1') for thing 'wemo:insight:insight1'
2022-01-21 23:30:35.829 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Received pair 'BinaryState':'0|1642803968|0|0|0|1209600|13|0|125088|125088' (service 'basicevent1') for thing 'wemo:insight:insight1'
```

But seeing the code again I can understand why nothing happens. It seems it literally compares the full string "0|1642803968|0|0|0|1209600|13|0|125088|125088" with "0"? Something is really broken, but may also have been broken to some extent before. Do you want me to try to figure out why it apparently worked with version 3.2?

Reply:
Besides what was mentioned above, I now did some more testing and comparison.

main branch:
```
2022-01-22 00:05:24.966 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-221734K120001F: Subscribing to service basicevent1...
2022-01-22 00:05:24.988 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-221734K120001F: Subscribing to service insight1...
2022-01-22 00:05:24.996 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'wemo:insight:insight1' changed from INITIALIZING to ONLINE
2022-01-22 00:05:25.016 [DEBUG] [ing.wemo.internal.WemoHandlerFactory] - Trying to create a handler for ThingType 'wemo:insight
2022-01-22 00:05:25.026 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service basicevent1 succeeded
2022-01-22 00:05:25.048 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service insight1 succeeded
2022-01-22 00:05:25.048 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Received pair 'BinaryState':'0|1642803968|0|0|0|1209600|13|0|0|0' (service 'basicevent1') for thing 'wemo:insight:insight1'
2022-01-22 00:05:25.079 [TRACE] [ng.wemo.internal.handler.WemoHandler] - State response '<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body>
<u:GetInsightParamsResponse xmlns:u="urn:Belkin:service:metainfo:1">
<InsightParams>0|1642803968|0|0|0|1209600|13|0|0|0.000000|8000</InsightParams>
</u:GetInsightParamsResponse>
</s:Body> </s:Envelope>' for device 'wemo:insight:insight1' received
2022-01-22 00:05:25.082 [TRACE] [ng.wemo.internal.handler.WemoHandler] - New state '0|1642803968|0|0|0|1209600|13|0|0|0.000000|8000' for device 'wemo:insight:insight1' received
2022-01-22 00:05:25.085 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Received pair 'InsightParams':'0|1642803968|0|0|0|1209600|13|0|0|0.000000|8000' (service 'insight1') for thing 'wemo:insight:insight1'
2022-01-22 00:05:25.088 [TRACE] [ng.wemo.internal.handler.WemoHandler] - New InsightParam binaryState 'OFF' for device 'wemo:insight:insight1' received
```

This branch:
```
2022-01-22 00:09:56.430 [DEBUG] [ing.wemo.internal.WemoHandlerFactory] - Trying to create a handler for ThingType 'wemo:insight
2022-01-22 00:09:56.434 [DEBUG] [ing.wemo.internal.WemoHandlerFactory] - Creating a WemoHandler for thing 'wemo:insight:insight1' with UDN 'Insight-1_0-000001K000000F'
2022-01-22 00:09:56.449 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Creating a WemoHandler for thing 'wemo:insight:insight1'
2022-01-22 00:09:56.476 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'wemo:insight:insight1' changed from UNINITIALIZED (HANDLER_MISSING_ERROR) to INITIALIZING
2022-01-22 00:09:56.497 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Initializing WemoHandler for UDN 'Insight-1_0-000001K000000F'
2022-01-22 00:09:56.503 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Polling job
2022-01-22 00:09:56.505 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'wemo:insight:insight1' changed from INITIALIZING to ONLINE
2022-01-22 00:09:56.622 [TRACE] [ng.wemo.internal.handler.WemoHandler] - State response '<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
<s:Body>
<s:Fault>
<faultcode>s:Client</faultcode>
<faultstring>UPnPError</faultstring>
<detail>
<UPnPError xmlns="urn:schemas-upnp-org:control-1-0">
<errorCode>-111</errorCode>
<errorDescription>Invalid Action</errorDescription>
</UPnPError>
</detail>
</s:Fault>
</s:Body>
</s:Envelope>
' for device 'wemo:insight:insight1' received
2022-01-22 00:09:56.629 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Checking WeMo GENA subscription for 'org.openhab.binding.wemo.internal.handler.WemoHandler@9cbfe1'
2022-01-22 00:09:56.633 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-000001K000000F: Subscribing to service basicevent1...
2022-01-22 00:09:56.644 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-000001K000000F: Subscribing to service insight...
2022-01-22 00:09:56.659 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service basicevent1 succeeded
2022-01-22 00:09:56.672 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Received pair 'BinaryState':'0|1642803968|0|0|0|1209600|13|0|0|0' (service 'basicevent1') for thing 'wemo:insight:insight1'
2022-01-22 00:09:56.676 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - State 'ON' for device 'wemo:insight:insight1' received
```

Notice:

```
2022-01-22 00:05:24.966 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-221734K120001F: Subscribing to service basicevent1...
2022-01-22 00:05:24.988 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-221734K120001F: Subscribing to service insight1...
2022-01-22 00:05:25.026 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service basicevent1 succeeded
2022-01-22 00:05:25.048 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service insight1 succeeded
```

vs.

```
2022-01-22 00:09:56.633 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-000001K000000F: Subscribing to service basicevent1...
2022-01-22 00:09:56.644 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - Setting up GENA subscription Insight-1_0-000001K000000F: Subscribing to service insight...
2022-01-22 00:09:56.659 [DEBUG] [ng.wemo.internal.handler.WemoHandler] - WeMo Insight-1_0-000001K000000F: Subscription to service basicevent1 succeeded
```

It seems that a "1" is missing when trying to subscribe to insights, i.e. should be "insight1", not "insight".

Reply:
I need to investigate a bit more on the state update. In the meantime, I have reverted the changes in Line 229 to
        if ("insight".equals(getThing().getThingTypeUID())) {

Reply:
FWIW, one of my WeMo Insights is still not working correctly after fce90f4c032841840b0ff17852e6f475907df699 because of this, so reopening comment. Subscription to **insight1** events should be okay now, but I'm only receiving **basicevent1** with binaryState.

Reply:
I will add a case switch before parsing, so insightParams and basicevent will be parsed correct for insight devices…

Reply:
Solved slightly different now...